### PR TITLE
Add git hooks

### PR DESCRIPTION
**Pull request by Anderssorby**
Add opt-in git hooks that runs tests before commit.

### CODE REVIEW

File: hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+#
+# An example hook script to verify what is about to be committed.
+# Called by "git commit" with no arguments.  The hook should
```
Comment (stianste) [inline hooks/pre-commit:4]: Does this mean that this will not run when running `git commit -m 'my commit'` for instance?

File: hooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# exit with non-zero status after issuing an appropriate message if
+# it wants to stop the commit.
+#
+# To enable this hook, rename this file to "pre-commit".
+
+if git rev-parse --verify HEAD >/dev/null 2>&1
+then
+	against=HEAD
```
Comment (stianste) [inline hooks/pre-commit:12]: Do we want to use 8 spaces here?

File: hooks/pre-commit (hunk 1 above)
Comment (Anderssorby) [reply hooks/pre-commit:4]: I guess. Let's test

**Review by stianste — APPROVED** [review]
LGTM.

File: hooks/pre-commit (hunk 2 above)
Comment (Anderssorby) [reply hooks/pre-commit:12]: Maybe not. It's just from the git template file

**Pull request merged**